CMakeLists.txt

```diff
@@ -1,52 +1,32 @@
 cmake_minimum_required(VERSION 3.2)
 
-list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake")
-
 project(FunctionalPlus VERSION 0.2.10)
 
-message(STATUS "===( ${PROJECT_NAME} ${PROJECT_VERSION} )===")
-option(FPLUS_USE_TOOLCHAIN "Use compiler flags from an external toolchain" OFF)
-option(FPLUS_BUILD_EXAMPLES "Build examples" ON)
-option(FPLUS_BUILD_UNITTEST "Build unit tests" OFF)
-option(FPLUS_UNITTEST_USE_CONAN "Use conan to get doctest dependency (for unit tests only)" OFF)
-
-message(STATUS "Building Unit Tests ${FPLUS_BUILD_UNITTEST}")
-message(STATUS "Use conan ${FPLUS_UNITTEST_USE_CONAN}")
-message(STATUS "Building examples ${FPLUS_BUILD_EXAMPLES}")
-
-if(NOT FPLUS_USE_TOOLCHAIN)
-  if (MSVC)
-    include(cmake/toolchain_msvc.cmake)
-  else()
-    include(cmake/toolchain.cmake)
-  endif()
+# ---- Warning guard ----
+
+# Protect dependents from this project's warnings if the guard isn't disabled
+set(FunctionalPlus_warning_guard SYSTEM)
+if(FunctionalPlus_INCLUDE_WITHOUT_SYSTEM)
+  set(FunctionalPlus_warning_guard "")
 endif()
 
+# ---- Declare library ----
+
 add_library(fplus INTERFACE)
 add_library(FunctionalPlus::fplus ALIAS fplus)
-target_include_directories(fplus INTERFACE
-  $<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>
-  )
+target_include_directories(
+    fplus
+    ${FunctionalPlus_warning_guard}
+    INTERFACE
+    "$<BUILD_INTERFACE:${CMAKE_CURRENT_LIST_DIR}/include>"
+)
+
+# CMake 3.2 doesn't know cxx_std_14 so instead we use cxx_variable_templates
+target_compile_features(fplus INTERFACE cxx_variable_templates)
```

Comment:
Why not increase to a suited CMake Version? :smile: 

@Dobiasd Are there any reasons to stick with v3.2?

Reply:
Not that I know of. :slightly_smiling_face: